### PR TITLE
UT for recent label cache fix (#369)

### LINKED ISSUE #371: Add libcalico-go UT for k8s backend label cache fix

### PR DESCRIPTION
**Pull request by neiljerram**
Fixes #371

### CODE REVIEW

**Comment by neiljerram** [comment]
This PR adds UT for #369, which was a fix for #368

**Comment by neiljerram** [comment]
@caseydavenport / @robbrockbank If you have time, I would appreciate your thoughts on this.

**Comment by neiljerram** [comment]
@caseydavenport @robbrockbank Even if the label cache might be destined for removal soon, I think the UT framework that this PR introduces will be useful for other things.

**Comment by fasaxc** [comment]
@neiljerram Assigning back to you to resolve conflicts etc.

**Comment by neiljerram** [comment]
@fasaxc Sure, will do.

**Comment by neiljerram** [comment]
Just rebased.

**Comment by neiljerram** [comment]
@fasaxc @robbrockbank Would you guys like to review this?  I'd like to get at least an initial view on it.

**Comment by neiljerram** [comment]
@fasaxc Added your syncer.Stop suggestion.  Also rebased up, which meant that I had to add new kubeAPI framework for HostConfigList.

@fasaxc / @robbrockbank Please could you take another look?  Will be great if we can merge this now.  (And I don't think there is any reason why we shouldn't!)

**Comment by neiljerram** [comment]
Fixed conflicts and squashed a bit.

File: lib/backend/k8s/syncer_test.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+				},
+			}
+			key := model.WorkloadEndpointKey{
+				Hostname:       "my-host-1",
+				OrchestratorID: "k8s",
+				WorkloadID:     "my-namespace.my-pod",
+				EndpointID:     "eth0",
+			}
+			getModelEndpoint := func() interface{} {
+				log.WithField("val", tc.state[key]).Info("state")
```
Comment (fasaxc) [inline lib/backend/k8s/syncer_test.go]: Think you need some synchronisation here to protect `tc.state`

Comment (neiljerram) [inline lib/backend/k8s/syncer_test.go]: I've added this now.  Please could you re-review?

**Pull request merged**